sql/catalyst/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala

```diff
@@ -79,7 +79,7 @@ private[sql] object ArrowUtils {
     case ArrowType.Null.INSTANCE => NullType
     case yi: ArrowType.Interval if yi.getUnit == IntervalUnit.YEAR_MONTH => YearMonthIntervalType()
     case di: ArrowType.Duration if di.getUnit == TimeUnit.MICROSECOND => DayTimeIntervalType()
-    case _ => throw QueryExecutionErrors.unsupportedDataTypeError(dt.toString)
+    case _ => throw QueryExecutionErrors.unsupportedDataTypeError(dt.asInstanceOf[DataType])
```

Comment:
hmm, `ArrowType` and `DataType` are unrelated types, this fails in runtime I guess.

Just in case, do we have a test for the case.

Reply:
Yeah, you're right.
I added test case and new error class `UNSUPPORTED_ARROWTYPE` to handle the ArrowType separately since I think they're technically different objects.